src/bin/lttng-sessiond/kernel.c

```diff
@@ -191,6 +193,227 @@ int kernel_create_channel(struct ltt_kernel_session *session,
 	return -1;
 }
 
+/*
+ * Compute the offset of the instrumentation byte in the binary based on the
+ * function probe location using the ELF lookup method.
+ *
+ * Returns 0 on success and set the offset out parameter to the offset of the
+ * elf symbol
+ * Returns -1 on error
+ */
+static
+int extract_userspace_probe_offset_function_elf(
+		struct lttng_userspace_probe_location *probe_location,
+		struct ltt_kernel_session *session, uint64_t *offset)
+{
+	int ret, fd;
+	const char *symbol = NULL;
+	struct lttng_userspace_probe_location_lookup_method *lookup = NULL;
+	enum lttng_userspace_probe_location_lookup_method_type lookup_method_type;
+
+	ret = 0;
```

Comment:
We could initialize `ret` at the declaration site.

Reply:
fixed